remediation-service/handler/remediation.go

```diff
@@ -0,0 +1,53 @@
+package handler
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/keptn/go-utils/pkg/api/models"
+	"github.com/keptn/go-utils/pkg/lib/v0_1_4"
+	"github.com/keptn/go-utils/pkg/lib/v0_2_0"
+)
+
+func ParseRemediationResource(resource *models.Resource) (*v0_1_4.Remediation, error) {
+	remediationData := &v0_1_4.Remediation{}
+	err := yaml.Unmarshal([]byte(resource.ResourceContent), remediationData)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse remediation.yaml: %w", err)
+	}
+
+	if remediationData.ApiVersion != remediationSpecVersion {
+		return nil, fmt.Errorf("remediation.yaml file does not conform to remediation spec %s", remediationSpecVersion)
+	}
+	return remediationData, nil
+}
+
+func GetNextAction(remediation *v0_1_4.Remediation, problemDetails v0_2_0.ProblemDetails, actionIndex int) (*v0_2_0.ActionInfo, error) {
+	rootCause := problemDetails.RootCause
+	problemTitle := problemDetails.ProblemTitle
+
+	var actions []v0_1_4.RemediationActionsOnOpen
+	for _, r := range remediation.Spec.Remediations {
+		if r.ProblemType == rootCause || r.ProblemType == problemTitle {
```

Comment:
would it make sense to first go through all remediations and check if a remediation for the root cause is available?
This way, having a remediation that matches the problem title would override a more specific one, if it is defined earlier in the shipyard. e.g.:

```
problemTitle = "response time degradation"
rootCause = "failure rate increase"

remediations:
  - problemType: "response time degradation"
    actionsOnOpen:
      - name: "some action"
      
  - problemType: "failure rate increase"
    actionsOnOpen:
      - name: "other action that would be better suited"
```

In this example, the service would determine that the correct action to take would be the first one (based on the problem title), while the other one that addresses the root cause will not be returned.

Reply:
good catch, will do it